src/Shared/Buffers.MemoryPool/PinnedBlockMemoryPool.cs

```diff
@@ -39,13 +42,44 @@ internal sealed class PinnedBlockMemoryPool : MemoryPool<byte>
     /// </summary>
     private bool _isDisposed; // To detect redundant calls
 
+    private readonly PinnedBlockMemoryPoolMetrics _metrics;
+
+    private long _currentMemory;
+    private long _evictedMemory;
+    private DateTimeOffset _nextEviction = DateTime.UtcNow.AddSeconds(10);
+
+    private uint _rentCount;
+    private uint _returnCount;
+
     private readonly object _disposeSync = new object();
 
+    private Action<object?, PinnedBlockMemoryPool>? _onPoolDisposed;
+    private object? _onPoolDisposedState;
+
     /// <summary>
     /// This default value passed in to Rent to use the default value for the pool.
     /// </summary>
     private const int AnySize = -1;
 
+    public PinnedBlockMemoryPool()
+        : this(NoopMeterFactory.Instance)
+    {
+    }
+
+    public PinnedBlockMemoryPool(IMeterFactory meterFactory)
```

Comment:
Maybe we should pass an options object so we can easily add future options

Reply:
How would you set the options? Just have a generic `MemoryPoolOptions` type that users can configure in DI? That sounds like something we could add in the future without breaking anything.